website/blog/2025-07-01-mlflow-prompt-evaluate/index.md

```diff
@@ -0,0 +1,435 @@
+
+Having a robust way to iterate on prompts and assess their accuracy becomes paramount in building production-ready GenAI applications.
+
+## MLflow Key Capabilities
+
+### MLflow Prompt Registry
+
+[MLflow Prompt Registry](https://mlflow.org/docs/latest/genai/prompt-version-mgmt/prompt-registry) addresses many of the challenges prompt engineers face today. With a Git-inspired set up, prompt engineers have a streamlined way to version, track, and reuse prompts. Let's look at a typical flow.
+
+**1. Create and Register Your Prompt:** Each prompt is registered as a **Prompt Object**. This becomes a versioned, paramaterized template that can be dynamically filled at runtime. The object will include a **name** (unique identifier), **template text**, **version #**, **commit message**, **metadata/tags** (optional), and **aliases**.  
+Register your prompt with -
+`mlflow.genai.register_prompt()`
+
+**2. Manage Aliases:** Aliases ensure prompts remain aligned with their intended use. For example, "Production" and "Staging" aliases tie them to their respective environments, simplifying traceability, orchestration, and governance of our prompts.
+
+**3. Load and Use Prompts:** We are now ready to load in our prompt and start using it with our GenAI applications, like our LLM-backed OCR tool.  
+Load in your prompt with -
+`mlflow.genai.load_prompt()`
+
+**4. Iterate on Prompts:** An important step in our process is improving and updating prompts. This can be done any number of times while moving toward the accuracy required, all while staying organized in a centralized repository for prompt management.  
+<br/>
+
+### MLflow Evaluate
+
+[MLflow Evaluate](https://mlflow.org/docs/latest/genai/eval-monitor/) is key in our example today as we iterate on and improve our OCR tool. Evaluation techniques within MLflow allow us to systematically asses the performance of both traditional ML models and LLMs by incorporating built-in and custom metrics in our model iteration.
+
+[Built-in metrics](https://mlflow.org/docs/latest/api_reference/python_api/mlflow.metrics.html) are a great way to get started. They are automatically computed based on your model type. The exact set will depend on your configurations, but for LLM and agent evaluation we are able to follow traditional NLP metrics like BLEU and ROUGE, and LLM-judged metrics that will assess outputs like accuracy, relevance, toxicity, and more.
+
+Custom metrics can enable additional validations for traditional models, but really come in handy for our GenAI and agentic workflows. Custom metric functions are passed into the `mlflow.evaluate` API. The function signature is flexible, allowing us to pass in things like request, response, expected_response, guidelines, trace, tool_calls, etc. This enables you to meet evaluation criteria and scoring thresholds to ensure the model is up to production standards.
```

Comment:
Can we rewrite this? Plz include
1. Why do we use custom metrics? (current solid)
2. What is a custom metric? (current could be improved)
3. Closing sentence (current is good content but confusing wording)